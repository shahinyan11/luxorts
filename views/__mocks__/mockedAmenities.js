export default {
  'Amina Heaney': {
    '5e066ad7-4194-4bd4-95ec-0d9715f8159b': {
      id: '5e066ad7-4194-4bd4-95ec-0d9715f8159b',
      description: 'Maiores porro nostrum dolorum sunt.',
      name: 'Nolan Kertzmann',
      status: 'active',
      category: null,
      checked: false,
    },
  },
  'Torrie Sawayn': {
    'c277033f-1683-4d2d-a409-d00b0bbe8201': {
      id: 'c277033f-1683-4d2d-a409-d00b0bbe8201',
      description: 'Nemo magnam dolores quam voluptatibus distinctio unde.',
      name: 'Stephine Rohan',
      status: 'active',
      category: null,
      checked: false,
    },
  },
};
