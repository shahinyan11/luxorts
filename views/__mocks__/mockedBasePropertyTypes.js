export default [
  {
    value: '51e35121-c27e-46fc-9feb-b51dab2490ee',
    label: 'repudiandae',
    description: 'description repudiandae',
  },
  {
    value: '995c5714-185d-4085-8512-690903334f6e',
    label: 'molestias',
    description: 'description molestias',
  },
];
