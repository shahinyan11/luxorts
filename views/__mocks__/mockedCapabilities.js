export default {
  APPROVED: {
    '11da7909-9f4c-45d6-a2a0-8a105b067ed2': {
      id: '11da7909-9f4c-45d6-a2a0-8a105b067ed2',
      description: 'Vel commodi ad incidunt repellendus a molestias delectus nam.',
      name: 'Tonya Wunsch',
      status: 'active',
      checked: false,
    },
    'b84c898e-2176-4ac8-9dfa-54186ccef6cb': {
      id: 'b84c898e-2176-4ac8-9dfa-54186ccef6cb',
      description: 'Maxime nostrum ex iste velit nulla incidunt.',
      name: 'Edwin Greenholt',
      status: 'active',
      checked: false,
    },
  },
};
