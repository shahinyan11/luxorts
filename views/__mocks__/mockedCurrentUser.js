export default {
  id: '5c5d814c-8e06-433d-b006-4119311585bd',
  email: 'evhenii.hryhoriev+10@rubygarage.org',
  publicId: 'c4toxHVi',
  verifiedStatus: 'unverified',
  userType: 'traveller_account',
  userProfile: {
    id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
    about: null,
    address: null,
    avatarUrls: {
      original: '/avatar-img-url',
    },
    dateOfBirth: '1994-01-01',
    drivingLicenseFirstName: 'Evhenii',
    drivingLicenseLastName: 'Test',
    firstName: 'Evhenii',
    gender: null,
    lastName: 'Test',
    phoneNumber: '+299999999',
    street: 'street',
    city: 'city',
    state: 'state',
    zipCode: 'zipCode',
    country: 'country',
  },
  userInvitation: [
    {
      id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
      created_at: '2022-06-23T06:32:55.588Z',
      email: '12@mail.ru',
      last_sent_at: '2022-06-23T06:32:55.474Z',
      resendable: true,
      status: 'pending',
      user_account_id: '7211ccc2-1fca-4799-b1e1-4005e9fc42be',
    },
  ],
};
