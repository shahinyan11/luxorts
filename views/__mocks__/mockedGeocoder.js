export default [
  {
    streetNumber: '240',
    street: '240 North 7th Avenue',
    city: 'Phoenix',
    state: 'Arizona',
    country: 'United States',
    zipCode: '85007',
    fullAddress: '240 N 7th Ave, Phoenix, AZ 85007, USA',
    longitude: -112.0825682,
    latitude: 33.4507026,
  },
];
