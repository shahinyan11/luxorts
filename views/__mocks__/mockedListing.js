export default {
  id: '0645baf3-945d-4122-8a0b-5ab3e95a9b3e',
  applicationStep: 'accomodation',
  status: 'in_draft',
  userAccount: { id: '6cbf2b68-4a26-41b5-bcd5-161cc37e1420', type: 'hostAccount' },
  listingPropertyType: {
    id: 'ea19f127-bc89-4799-bdbb-346126ac3845',
    typeOfPlace: 'private_place',
    propertyType: {
      id: '51e35121-c27e-46fc-9feb-b51dab2490ee',
      name: 'repudiandae',
      description: 'description repudiandae',
    },
  },
  listingLocation: {
    id: '2b4032ec-a469-4489-b7dc-536d8f9552eb',
    country: 'United States',
    city: 'Phoenix',
    state: 'Arizona',
    street: 'West Maryland Avenue',
    apartmentNumber: '12',
    zipCode: '12345',
    longitude: '-73.97447621784085',
    latitude: '40.764468117826794',
  },
  listingDescription: null,
  listingPricing: {
    pricePerDay: 64,
  },
};
