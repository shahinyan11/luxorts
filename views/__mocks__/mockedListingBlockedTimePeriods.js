export default ['25/07/2022', '26/07/2022', '27/07/2022'];
