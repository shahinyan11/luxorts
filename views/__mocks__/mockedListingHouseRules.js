export default {
  APPROVED: {
    '943a4405-32c8-4c6e-9251-b9a3eef80e7a': {
      id: '943a4405-32c8-4c6e-9251-b9a3eef80e7a',
      description: 'Fugiat inventore totam numquam porro accusantium recusandae.',
      name: 'Ta Beier',
      status: 'active',
      checked: false,
      selectedId: undefined,
    },
    'ddf1ae63-b4eb-49c0-8130-1c5cffb14af1': {
      id: 'ddf1ae63-b4eb-49c0-8130-1c5cffb14af1',
      description: 'Tenetur eligendi similique nam repudiandae incidunt.',
      name: 'Kellie Zieme',
      status: 'active',
      checked: false,
      selectedId: undefined,
    },
  },
};
