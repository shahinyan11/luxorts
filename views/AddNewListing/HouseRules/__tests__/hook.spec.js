import { renderHook } from '@testing-library/react-hooks';

import ROUTES from 'constants/routes';
import { NOT_APPROVED } from 'constants';

import { createRoute } from 'utils/createRouteHelpers';
import checkUserVerification from 'utils/auth/checkUserVerification';

import {
  fetchListing,
  fetchListingBaseHouseRules,
  fetchListingHouseRules,
} from 'state/concepts/listings/actions';
import { dataDeleteEntity } from 'state/data/actions';
import { store } from '__mocks__/react-redux';
import mockedListing from 'views/__mocks__/mockedListing';
import mockedListingHouseRules from 'views/__mocks__/mockedListingHouseRules';

import useContainer, { getInitialProps } from '../hook';

jest.mock('utils/auth/checkUserVerification', () => jest.fn(() => true));
jest.mock('hooks/useParametricSelector', () => jest.fn(() => false));
jest.mock('state/concepts/listings/selectors', () => ({
  listingSelector: jest.fn(() => mockedListing),
  listingHouseRulesSelector: jest.fn(() => mockedListingHouseRules),
}));
jest.mock('utils/listing/checkListingCompleteness');

const mockedListingId = 'listingId';
const mockedRouter = {
  query: { listingId: mockedListingId },
};
jest.mock('next/router', () => ({
  useRouter: jest.fn(() => mockedRouter),
}));

const mockedFormik = {
  values: {
    houseRules: mockedListingHouseRules,
    listingId: mockedListingId,
    additional: {
      name: '',
    },
  },
  setValues: jest.fn(),
  setFieldValue: jest.fn(),
  handleSubmit: jest.fn(),
};
jest.mock('formik', () => ({
  useFormik: jest.fn(() => mockedFormik),
}));

describe('HouseRules useContainer hook', () => {
  let result = null;

  beforeEach(() => {
    jest.clearAllMocks();

    ({ result } = renderHook(useContainer));
  });

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  it('handleSubmit method should sets redirectRoute before submit', async () => {
    await result.current.handleSubmit();

    expect(mockedFormik.setFieldValue).toHaveBeenCalledWith(
      'redirectRoute',
      createRoute({
        pathname: ROUTES.ADD_NEW_LISTING.RESERVATION_PREFERENCES.PATH,
        id: mockedListingId,
      }),
    );
    expect(mockedFormik.handleSubmit).toHaveBeenCalled();
  });

  it('handleAddNotApprovedHouseRule method should resets redirectRoute and sets message before submit', async () => {
    await result.current.handleAddNotApprovedHouseRule();

    expect(mockedFormik.setValues).toHaveBeenCalledWith({
      ...mockedFormik.values,
      redirectRoute: null,
      message: {
        description: {
          id: 'listing.houseRules.additional.success',
        },
      },
    });
    expect(mockedFormik.handleSubmit).toHaveBeenCalled();
  });

  it('handleRemoveNotApprovedHouseRuleCreator should calls formik.setFieldValue', () => {
    result.current.handleRemoveNotApprovedHouseRuleCreator('id')();

    expect(mockedFormik.setFieldValue).toHaveBeenCalledWith(
      `houseRules.${NOT_APPROVED}.id.checked`,
      false,
    );
  });

  describe('getInitialProps method', () => {
    const ctx = {
      store,
      query: {
        listingId: mockedListingId,
      },
      req: {},
    };

    it('should dispatches all actions when user is verified and listingId is present', async () => {
      await getInitialProps(ctx);

      expect(store.dispatch).toHaveBeenCalledTimes(5);

      expect(store.dispatch).toHaveBeenNthCalledWith(1, dataDeleteEntity('houseRule'));
      expect(store.dispatch).toHaveBeenNthCalledWith(2, dataDeleteEntity('listingHouseRule'));
      expect(store.dispatch).toHaveBeenNthCalledWith(
        3,
        fetchListingBaseHouseRules(mockedListingId),
      );
      expect(store.dispatch).toHaveBeenNthCalledWith(4, fetchListing(mockedListingId));
      expect(store.dispatch).toHaveBeenNthCalledWith(5, fetchListingHouseRules(mockedListingId));
    });

    it('should dispatches only general actions when user is verified and listingId isn`t present', async () => {
      await getInitialProps({
        ...ctx,
        query: {},
      });

      expect(store.dispatch).toHaveBeenCalledTimes(3);

      expect(store.dispatch).toHaveBeenNthCalledWith(1, dataDeleteEntity('houseRule'));
      expect(store.dispatch).toHaveBeenNthCalledWith(2, dataDeleteEntity('listingHouseRule'));
      expect(store.dispatch).toHaveBeenNthCalledWith(
        3,
        fetchListingBaseHouseRules(mockedListingId),
      );
    });

    it('shouldn`t dispatches any actions when user isn`t verified', async () => {
      checkUserVerification.mockReturnValueOnce(false);

      await getInitialProps(ctx);

      expect(store.dispatch).not.toHaveBeenCalled();
    });

    it('should calls whenComplete callback when is server', async () => {
      await getInitialProps(ctx);

      expect(store.logicMiddleware.whenComplete).toHaveBeenCalled();
    });

    it('shouldn`t calls whenComplete callback when isn`t server', async () => {
      await getInitialProps({
        ...ctx,
        req: null,
      });

      expect(store.logicMiddleware.whenComplete).not.toHaveBeenCalled();
    });
  });
});
