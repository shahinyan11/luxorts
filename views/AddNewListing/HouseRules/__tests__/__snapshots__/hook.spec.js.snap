// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseRules useContainer hook matches snapshot 1`] = `
Object {
  "approvedHouseRules": Object {
    "943a4405-32c8-4c6e-9251-b9a3eef80e7a": Object {
      "checked": false,
      "description": "Fugiat inventore totam numquam porro accusantium recusandae.",
      "id": "943a4405-32c8-4c6e-9251-b9a3eef80e7a",
      "name": "Ta Beier",
      "selectedId": undefined,
      "status": "active",
    },
    "ddf1ae63-b4eb-49c0-8130-1c5cffb14af1": Object {
      "checked": false,
      "description": "Tenetur eligendi similique nam repudiandae incidunt.",
      "id": "ddf1ae63-b4eb-49c0-8130-1c5cffb14af1",
      "name": "Kellie Zieme",
      "selectedId": undefined,
      "status": "active",
    },
  },
  "formik": Object {
    "handleSubmit": [MockFunction],
    "setFieldValue": [MockFunction],
    "setValues": [MockFunction],
    "values": Object {
      "additional": Object {
        "name": "",
      },
      "houseRules": Object {
        "APPROVED": Object {
          "943a4405-32c8-4c6e-9251-b9a3eef80e7a": Object {
            "checked": false,
            "description": "Fugiat inventore totam numquam porro accusantium recusandae.",
            "id": "943a4405-32c8-4c6e-9251-b9a3eef80e7a",
            "name": "Ta Beier",
            "selectedId": undefined,
            "status": "active",
          },
          "ddf1ae63-b4eb-49c0-8130-1c5cffb14af1": Object {
            "checked": false,
            "description": "Tenetur eligendi similique nam repudiandae incidunt.",
            "id": "ddf1ae63-b4eb-49c0-8130-1c5cffb14af1",
            "name": "Kellie Zieme",
            "selectedId": undefined,
            "status": "active",
          },
        },
      },
      "listingId": "listingId",
    },
  },
  "handleAddNotApprovedHouseRule": [Function],
  "handleRemoveNotApprovedHouseRuleCreator": [Function],
  "handleSubmit": [Function],
  "isAddButtonDisabled": true,
  "isLoading": false,
  "notApprovedHouseRules": Array [],
  "onBackRoute": "/add-new-listing/description/listingId",
  "showSkeleton": false,
  "stepStatus": "accomodation",
}
`;
